xmtp_api_d14n/src/protocol/traits/cursor_store.rs

```diff
@@ -67,28 +68,12 @@ pub trait CursorStore: Send + Sync {
         })
     }
 
-    /// A temporary function to get the latest cursor for
-    /// a topic & originator. it may be missing updates.
-    fn latest_maybe_missing_per(
+    /// Get the latest cursor for multiple topics at once.
+    /// Returns a HashMap mapping each topic to its GlobalCursor.
+    fn latest_for_topics(
         &self,
-        topic: &Topic,
-        originator: &[&OriginatorId],
-    ) -> Result<GlobalCursor, CursorStoreError>;
-
-    /// Temporary until reliable streams
-    fn latest_maybe_missing(
-        &self,
-        topic: &Topic,
-        originator: &OriginatorId,
-    ) -> Result<Cursor, CursorStoreError> {
-        let sid = self
-            .latest_maybe_missing_per(topic, &[originator])?
-            .get(originator);
-        Ok(Cursor {
-            originator_id: *originator,
-            sequence_id: sid,
-        })
-    }
+        topics: &[&Topic],
```

Comment:
Can this be defined as `&mut dyn Iterator<Item = &Topic>` to avoid unneeded allocations for those impls that don't need them while those that do can collect into a vec?

Reply:
It works. Updated